### PR TITLE
Fix - getEntitiesRestrictCriteria() returns invalid SQL criterion when no session is active

### PR DESCRIPTION
**Pull request by RomainLvr**
## Checklist before requesting a review

*Please delete options that are not relevant.*

- [x] I have read the CONTRIBUTING document.
- [x] I have performed a self-review of my code.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] This change requires a documentation update.

## Description

- It fixes !43370 & https://github.com/pluginsGLPI/fields/pull/1179
- Here is a brief description of what this PR does

When `getEntitiesRestrictCriteria()` is called with no active session, no CLI context, and no cron context, the function falls through all conditions without an else branch. This leaves `$value` as an empty string `''`, producing the invalid SQL criterion `entities_id = ''` on an integer column and triggering MySQL warning 1292 (Truncated incorrect DECIMAL value).

This fix adds a missing else branch that returns `[new QueryExpression('false')]` when no valid context is available, denying all access instead of generating invalid SQL. If there is no session, no results should be returned.

Fixes all callers of `getEntitiesRestrictCriteria()` that are affected by this edge case (e.g. [pluginsGLPI/fields#1179](https://github.com/pluginsGLPI/fields/pull/1179)).

### CODE REVIEW

**Review by Rom1-B — CHANGES_REQUESTED** [review]
Please add tests

**Comment by RomainLvr** [comment]
The initial problem is that `isCommandLine()` relies on the `PHP_SAPI` constant, which is always set to ‘cli’ in PHPUnit. The solution we found: extract the call to `isCommandLine() || Session::isCron()` into a protected method called `isPrivilegedContext()`, which we override in a test using an anonymous class.

File: src/DbUtils.php [hunk 1]
```diff
@@ -52,7 +52,7 @@
  *
  * @since 9.2
  */
-final class DbUtils
+class DbUtils
```
Comment (Rom1-B) [inline src/DbUtils.php]: The class must be final
```suggestion
final class DbUtils
```
As we discussed in person, you can get away with using a global variable in `isCommandLine()`

**Review by Rom1-B — CHANGES_REQUESTED** [review]

File: src/DbUtils.php (hunk 1 above)
Comment (trasher) [reply src/DbUtils.php]: Not sure wa can change class to final in a bugfixes release.

**Review by Rom1-B — APPROVED** [review]

**Review by AdrienClairembault — APPROVED** [review]

**Review by trasher — APPROVED** [review]

**Review by cedric-anne — APPROVED** [review]
